packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.C

```diff
@@ -941,17 +949,20 @@ namespace Ioex {
     }
   }
 
-  void ParallelDatabaseIO::get_step_times_nl()
+  std::vector<double> ParallelDatabaseIO::internal_get_step_times_nl(bool setRegionTimeSteps)
```

Comment:
does this need to exist (_nl version) and does non-nl version need to acquiere a lock...

Reply:
Lines 1030 and 1035 are where this is called from and those are both _nl versions of calling functions. I would assume that since they are _nl functions, then there is no need to acquire a lock. Should I rename it to a non _nl version ?

Reply:
It is probably best to merge this and then go through and look at all the locks / _nl functions and see what is needed...  Too many locks shouldnt' affect anything (and thread-safe not run that much anyway)